src/fu-device.c

```diff
@@ -50,6 +50,8 @@ typedef struct {
 	guint				 remove_delay;	/* ms */
 	FwupdStatus			 status;
 	guint				 progress;
+	guint				 order;
+	FuDeviceChildFlags		 child_flags;
```

Comment:
Maybe child_behavior if you don't like flags?

Reply:
Or child_priority?

Reply:
I think sometimes when naming things is hard you need to re-asses what that "thing" represents. In this case we have a whole lot of extra complexity for a "do this before that", although it does allow us to add other things in the future like `ONLY_UPGRADABLE_IF_PARENT_IS` easy. I think maybe I need to bite the bullet and just add `FuDeviceChildFlags` to `fu_device_add_parent_guid` and store it in a struct in the array.

Reply:
More on this: logically, who gets to decide the order of the install tasks, the child or the parent? If it's the former then it could be one child wants to before the parent, and one after (which is fine) but in most cases it's the parent that's going to be saying "queue all the children first to avoid multiple restarts" or "you have to update the main FuDevice before the children". I think we can assume the parent-child relationship is by agreement, it just defines how we structure the API.